lumigator/python/mzai/backend/backend/api/routes/health.py

```diff
@@ -41,24 +42,63 @@ def get_job_metadata(job_id: UUID) -> JobSubmissionResponse:
     except json.JSONDecodeError as e:
         loguru.logger.error(f"JSON decode error: {e}")
         loguru.logger.error(f"Response text: {resp.text}")
+        raise HTTPException(
+            status_code=HTTPStatus.INTERNAL_SERVER_ERROR, detail="Invalid JSON response"
+        ) from e
+
+
+@router.get("/jobs/{job_id}/logs")
+def get_job_logs(job_id: UUID) -> JobLogsResponse:
+    resp = requests.get(urllib.parse.urljoin(settings.ray_jobs, f"{job_id}/logs"))
+
+    if resp.status_code == HTTPStatus.NOT_FOUND:
+        raise HTTPException(
+            status_code=HTTPStatus.NOT_FOUND,
+            detail=f"Job logs for ID: {job_id} not found - {resp.text or ''}",
+        )
+    elif resp.status_code == HTTPStatus.BAD_REQUEST:
+        raise HTTPException(
+            status_code=HTTPStatus.BAD_REQUEST,
+            detail=f"Job ID: {job_id} not submitted via the Ray API - {resp.text or ''}",
+        )
+    elif resp.status_code != HTTPStatus.OK:
+        loguru.logger.error(f"Unexpected status code getting job logs text: {resp.status_code}")
+        loguru.logger.error(f"Upstream job metadata response: {resp}")
         raise HTTPException(
             status_code=HTTPStatus.INTERNAL_SERVER_ERROR,
-            detail="Invalid JSON response"
+            detail=f"Unexpected error getting job logs for ID: {job_id} - {resp.text or ''}",
+        )
+
+    try:
+        metadata = json.loads(resp.text)
+        return JobLogsResponse(**metadata)
+    except json.JSONDecodeError as e:
+        loguru.logger.error(f"JSON decode error: {e}")
+        loguru.logger.error(f"Response text: {resp.text}")
+        raise HTTPException(
+            status_code=HTTPStatus.INTERNAL_SERVER_ERROR, detail="Invalid JSON response"
         ) from e
 
+
 @router.get("/jobs/")
 def get_all_jobs() -> list[JobSubmissionResponse]:
-    resp = requests.get(f"{settings.RAY_DASHBOARD_URL}/api/jobs/")
-    if resp.status_code == 200:
-        try:
-            metadata = json.loads(resp.text)
-            submissions: list[JobSubmissionResponse] = [
-                JobSubmissionResponse(**item) for item in metadata
-            ]
-            return submissions
-        except json.JSONDecodeError as e:
-            print(f"JSON decode error: {e}")
-            print(f"Response text: {resp.text}")
-            return {"error": "Invalid JSON response"}
-    else:
-        return {"error": f"HTTP error {resp.status_code}"}
+    resp = requests.get(settings.ray_jobs)
+    if resp.status_code != HTTPStatus.OK:
+        loguru.logger.error(f"Unexpected status code getting job logs text: {resp.status_code}")
+        loguru.logger.error(f"Upstream job metadata response: {resp}")
+        raise HTTPException(
+            status_code=HTTPStatus.INTERNAL_SERVER_ERROR,
+            detail=f"Unexpected error getting job logs - {resp.text or ''}",
```

Comment:
```suggestion
            detail=f"Unexpected error getting jobs",
```

Reply:
Ok

Reply:
Was this missed or was there consensus against changing it?